builder/parser/parser.go

```diff
@@ -61,6 +61,7 @@ func init() {
 		command.Entrypoint: parseMaybeJSON,
 		command.Expose:     parseStringsWhitespaceDelimited,
 		command.Volume:     parseMaybeJSONToList,
+		command.Arg:        parseNameOrNameVal,
```

Comment:
yes, that is correct and this is not an oversight. 

The check for allowing only one argument per line is done in the dispatcher function for `ARG` https://github.com/docker/docker/pull/15182/files#diff-9ead3fedd50f7ee2684c132d647703b6R601. I have added a test for the same as well (https://github.com/docker/docker/pull/15182/files#diff-64d1c3a97fd8b1abf170680883e81aaaR6044)

`parseNameOrNameVal` is the new generic function that adheres to the following parsing rules:
- it can parse one or more `name=value` pairs
- it can parse one or more `name=` words (for the case when default value is not specified)

I needed a new function as this function is slightly stricter than the `parseEnv` function (that calls the `parseNameVal`) in a way that it doesn't allow `ARG name value` format at all. Internally I do make use of the common logic for parsing `words in a statement` that existed for `parseEnv`. I have extracted that logic into `parseWords` utility function.

Hope this clarifies.